cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -75,6 +77,16 @@
 public class PluginInstantiator implements Closeable {
   private static final Logger LOG = LoggerFactory.getLogger(PluginInstantiator.class);
 
+  // used for setting defaults of string and non-string macro-enabled properties at config time
+  private static final Map<String, Class> PROPERTY_TYPES = ImmutableMap.<String, Class>builder()
+    .put("boolean", boolean.class)
+    .put("int", int.class)
+    .put("long", long.class)
+    .put("float", float.class)
+    .put("double", double.class)
+    .put("string", String.class)
```

Comment:
Just talked to @sagarkapare and he said they are supported. I'll add them, thanks